### ISSUE TITLE
classification task using softmax

### ISSUE THREAD

**Issue by KisaragiDev** [kisaragiRY/decoding-DG#39]
> use softmax regression to solve 4 classes classification problem
* [x] classes visualization
   * need upsampling? 
* [x] util: a softmax function
* [ ] base decoder
     * [ ] fit method
     * [ ] predict method
     * [ ] load method
     * [ ] evaluate method
* [ ] decoder: softmax regression based on gradient decent.
   * [ ] loss function method
   * [ ] derivatives method
* [x] implementation and evaluate 

## reference
1.  code: [Softmax Regression in Python: Multi-class Classification](https://towardsdatascience.com/softmax-regression-in-python-multi-class-classification-3cb560d90cb2)
2. math : [多元逻辑回归 · 数学推导过程及代码实现完全解析](https://blog.csdn.net/weixin_44780709/article/details/118082930)
